packages/providers/src/provider.ts

```diff
@@ -752,7 +752,7 @@ export default class Provider {
     let encoded: undefined | Uint8Array;
     if (predicateData && predicate.types) {
       const abiCoder = new AbiCoder();
-      encoded = abiCoder.encode(predicate.types, predicateData);
+      encoded = abiCoder.encode(predicate.types, predicateData as InputValue[]);
```

Comment:
Shouldn't this be `InputValue<T>[]`? _(see method signature on line `735`)._

Does it make any difference?

```suggestion
      encoded = abiCoder.encode(predicate.types, predicateData as InputValue<T>[]);
```

Reply:
Oh, my bad. You are right.

It looks like `values: InputValue[]` is expected.


https://github.com/FuelLabs/fuels-ts/blob/f521146c328a7fb2c98679ec3f0c9aa6df2f684f/packages/abi-coder/src/abi-coder.ts#L122